### PR TITLE
Properly handle incomplete HTML fragments

### PR DESCRIPTION
**Pull request by cmer**
# Type of PR (feature, enhancement, bug fix, etc.)

Bug Fix

## Description

Properly handle incomplete HTML fragments that would normally be stripped out by Nokogiri. For example, `<tr><td></td></td>` is invalid by itself because it is not wrapped in a `table` element. This PR detects these instances and handle them gracefully.

Fixes # (issue)

## Why should this be added

Ability to morph parts of a table is important.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] Checks (StandardRB & Prettier-Standard) are passing
- [x] This is not a documentation update

Please note that the best way to suggest changes or updates to the [documentation](https://docs.stimulusreflex.com) is to [join Discord](https://discord.gg/stimulus-reflex) and leave a note in the #docs channel. Any documentation updates posted as PRs cannot be accepted at this time. :heart:

### CODE REVIEW

File: lib/stimulus_reflex/html/document.rb [hunk 1]
```diff
@@ -7,20 +7,28 @@ class Document
...
+        end.to_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT)
       end
+      alias_method :to_html, :outer_html
 
       def inner_html
-        @document.root.inner_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT)
+        if @document.is_a?(Nokogiri::HTML::DocumentFragment)
+          # `inner_html` on a DocumentFragment returns the outer HTML tag, so we need to strip it out manually.
+          # There might be a cleaner way to do this?
+          @document.inner_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT).sub(/^<([^>]*)>/, "").sub(/<\/[^>]*>$/, "")
```
[the remark this answers is not in the document]
Comment (cmer) [reply lib/stimulus_reflex/html/document.rb:23]: Yeah. I couldn't find a clean way. I didn't spend a whole lot of time on it, but I agree it didn't feel great regexp'ing this. But it works. Kind of odd that `inner_html` returns the `outer_html`...

**Comment by marcoroth** [comment]
Hey @cmer, I pushed up a commit which merges your work with a previous state I had in #601. Would you mind testing this pull-request against your app to see if it still works as expected. Thank you!

**Comment by cmer** [comment]
I smoke-tested it in the browser and it seems to work, indeed! @marcoroth

File: lib/stimulus_reflex/html/document.rb [hunk 2]
```diff
@@ -7,20 +7,22 @@ class Document
...
 
       def inner_html
-        @document.root.inner_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT)
+        document_element ? document_element.inner_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT) : ""
       end
 
       def initialize(html)
-        @document = Nokogiri::HTML5::Document.parse(html.to_s)
+        @initial_html = html.to_s
+        @document = parsing_class.parse(@initial_html)
```
Comment (marcoroth) [inline lib/stimulus_reflex/html/document.rb:25]: I think we can even bring this back now
```suggestion
        @document = Nokogiri::HTML5::Document.parse(html.to_s)
```

**Comment by marcoroth** [comment]
Thank you @cmer!

**Review by marcoroth — APPROVED** [review]

**Pull request merged**